cuqi/randomvariable/_randomvariable.py

```diff
@@ -0,0 +1,191 @@
+                    "distribution is defined as a variable, e.g. x = Gaussian(0, 1) or provide "
+                    "a name, e.g. Gaussian(0, 1, name='x')"
+                )
+        
+        self._distributions = distributions
+        """ The distribution from which the random variable originates. """
+
+        if tree is None:
+            if len(distributions) > 1:
+                raise ValueError("Cannot create random variable from multiple distributions")
+            tree = RandomVariableNode(next(iter(distributions)).name)
+        
+        self._tree = tree
+        """ The tree representation of the random variable. """
+
+    def __call__(self, *args, **kwargs) -> Any:
+        """ Evaluate random variable at a given parameter value. """
+        if args and kwargs:
+            raise ValueError("Cannot pass both positional and keyword arguments to RandomVariable")
+        
+        if args:
+            kwargs = self._parse_args_add_to_kwargs(args, kwargs)
+
+        return self._tree(**kwargs)
+    
+    def sample(self):
+        """ Sample random variable. """
+        return self(**{distribution.name: distribution.sample() for distribution in self._distributions})
+         
+    @property
+    def parameter_names(self) -> str:
+        """ Name of the parameter that the random variable can be evaluated at. """
+        return [distribution.name for distribution in self._distributions]
+    
+    @property
+    def _non_default_args(self) -> List[str]:
+        """List of non-default arguments to distribution. This is used to return the correct
+        arguments when evaluating the random variable.
+        """
+        return self.parameter_names
+    
+    def _parse_args_add_to_kwargs(self, args, kwargs) -> dict:
+        """ Parse args and add to kwargs if any. Arguments follow self.parameter_names order. """
+        if len(args) != len(self.parameter_names):
+            raise ValueError(f"Expected {len(self.parameter_names)} arguments, got {len(args)}")
+        
+        # Add args to kwargs
+        for arg, name in zip(args, self.parameter_names):
+            kwargs[name] = arg
+        
+        return kwargs
+
+    def __repr__(self):
+        # Create strings for parameter name ~ distribution pairs
+        parameter_strings = [f"{name} ~ {distribution}" for name, distribution in zip(self.parameter_names, self._distributions)]
+        # Join strings with newlines
+        parameter_strings = "\n".join(parameter_strings)
+        # Add initial newline and indentations
+        parameter_strings = "\n".join(["\t"+line for line in parameter_strings.split("\n")])
+        # Print parameter strings with newlines
+        return (f"RandomVariable\n"
+                f"Distributions: \n{parameter_strings}\n"
+                f"Transformations: {self._tree}")
+   
+    def _apply_operation(self, operation, other=None) -> 'RandomVariable':
```

Comment:
`operation` seems to be a reserved keyword in python. I got a bit confused with the highlight then realized it is an input parameter

Reply:
I do not see this in my IDE. can you give some more details?